core/rubygems/version.rbs

```diff
@@ -167,7 +167,7 @@ module Gem
     # -->
     # True if the `version` string matches RubyGems' requirements.
     #
-    def self.correct?: (_ToS version) -> bool
+    def self.correct?: (_ToS & Object version) -> bool
```

Comment:
Technically `Object` also has a `.to_s` defined on it. But I think having `_ToS & Object` makes it clear that the `version` object should have a custom `.to_s` defined.

Reply:
type alias `str` for `String | _ToS` /halfjoke

anyways, would it be
```suggestion
    def self.correct?: (_ToS & Kernel version) -> bool
```
rather?

Reply:
Yeah that's probably better.